File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        Boolean allZero = true;
        for (int elem : scoville) {
            if (elem > 0) allZero = false;
            pq.offer(elem);
        }
        
        if (allZero) return -1;
        
        int cnt = 0;
        while (pq.peek() < K) {
            if (pq.size() == 1) return -1;
            
            int newFood = pq.poll() + pq.poll() * 2;
            pq.offer(newFood);
            cnt++;
        }
        
        return cnt;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    
    priority_queue<int, vector<int>, greater<int>> pq;
    for (int elem : scoville) pq.push(elem);
    
    // 더 이상 섞을 수 없을 때까지
    while (pq.size() != 1) {
        if (pq.top() >= K) break;
        
        int first = pq.top();
        pq.pop();
        
        int second = pq.top();
        pq.pop();
        
        pq.push(first + second * 2);
        answer++;
    }
    
    if (pq.top() < K) return -1;
    else return answer;
}